api/v2/api.go

```diff
@@ -508,7 +508,50 @@ func (api *API) getSilencesHandler(params silence_ops.GetSilencesParams) middlew
 		sils = append(sils, &silence)
 	}
 
-	return silence_ops.NewGetSilencesOK().WithPayload(sils)
+	active := open_api_models.GettableSilences{}
```

Comment:
Instead of 3 intermediary slices, you could use `sils` directly and expand the less function of `sort.Slice` to sort on `GettableSilence.Status.State` too.

Reply:
I was originally thinking about this, but was concerned that it might make the code less readable.  So I added a comment to hopefully make it a bit more clear what the intentions are for the sorting.